File: Jeonghak/Programmers/타겟넘버.cpp
#include <string>
#include <vector>

using namespace std;

int answer = 0;

void dfs(vector<int> numbers, int target, int currValue, int depth){
    if(depth == numbers.size()-1){
        if(target == currValue){
            answer++;
        }
        return;
    }
    
    dfs(numbers, target, currValue + numbers[depth+1], depth+1);
    dfs(numbers, target, currValue - numbers[depth+1], depth+1);
}    


int solution(vector<int> numbers, int target) {    
    dfs(numbers, target, numbers[0], 0);
    dfs(numbers, target, -numbers[0], 0);
    return answer;
}

File: jokercsi/Programmers/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    leaves = [0]
    
    for i in numbers:
        tmp = []
        for parent in leaves:
            tmp.append(parent + i)
            tmp.append(parent - i)
        leaves = tmp
        
    for leaf in leaves:
        if leaf == target:
            answer += 1
    
    return answer
